packages/shared/src/components/modals/common/Modal.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { ReactElement } from 'react';
+import ReactModal from 'react-modal';
+import classNames from 'classnames';
+import { ModalHeader } from './ModalHeader';
+import { ModalBody } from './ModalBody';
+import { ModalFooter } from './ModalFooter';
+import { ModalSidebar } from './ModalSidebar';
+import { ModalKind, ModalPropsContext, ModalSize } from './types';
+
+export type ModalProps = ReactModal.Props & {
+  children: React.ReactNode;
+  kind?: ModalKind;
+  size?: ModalSize;
+};
+
+export function Modal({
+  className,
+  overlayClassName,
+  children,
+  kind = ModalKind.FlexibleCenter,
+  size = ModalSize.Medium,
+  onRequestClose,
+}: ModalProps): ReactElement {
+  const modalKindToOverlayClassName = {
+    [ModalKind.FixedCenter]: 'justify-center',
+    [ModalKind.FlexibleCenter]: 'justify-center',
+    [ModalKind.FlexibleTop]: '',
+  };
+  const modalKindToClassName = {
+    [ModalKind.FixedCenter]: '',
+    [ModalKind.FlexibleCenter]: 'h-[40rem]',
+    [ModalKind.FlexibleTop]: classNames(
+      'mt-10',
+      size === ModalSize.Medium && 'mt-20',
+      size === ModalSize.Large && 'mt-14',
+    ),
+  };
+  const modalSizeToClassName = {
```

Comment:
responsiveness support is missing, but the styles for the desktop sizes are here

Reply:
I am working on it, if we merge this PR, I will open a new one with that